src/windows/views/timeline_webview.py

```diff
@@ -1367,62 +1383,51 @@ def Copy_Triggered(self, action, clip_ids, tran_ids):
             elif action == MENU_COPY_KEYFRAMES_CONTRAST:
                 self.copy_transition_clipboard[tran_id]['contrast'] = tran.data['contrast']
 
+        # Move the copied part to the left side to simple paste later
+        if position_diff > 0 or layer_diff > 0:
+            print("  !!!!! DEBUG: fixing position of clipboard block: %d %d" % (position_diff, layer_diff))
+            # Adjust the copy block position and track
+            for clip_id in clip_ids:
+                self.copy_clipboard[clip_id]['position'] -= position_diff
+                self.copy_clipboard[clip_id]['layer'] -= layer_diff
+            for tran_id in tran_ids:
+                self.copy_transition_clipboard[tran_id]['position'] -= position_diff
+                self.copy_transition_clipboard[tran_id]['layer'] -= layer_diff
+
```

Comment:
This approach is never going to work, I'm afraid, because of the way layers work. The old code was really confusingly written — borderline misleading, now that I'm looking at it — so I can see where it steered you wrong. But, unfortunately, it did.

Layer numbers aren't _coordinates_ in any real sense. They're just indexes, and they aren't guaranteed to be either contiguous or evenly spaced. IOW, the layers for a project could look like this:
```json
 "layers": [
  {
   "number": 100,
   "id": "L1",
   "label": "",
   "lock": false
  },
  {
   "number": 110,
   "id": "L2",
  },
  {
   "number": 3000000,
   "id": "L3",
  },
  {
   "number": 3000001,
   "id": "L3",
  }]
```

The **only** guarantee is that each higher layer number will be greater than all of the ones below it.

So, you can't do _math_ on layer numbers in order to move things between layers. It just doesn't work. It **can't** work. The old code was using the layer number "offsets" to to make all of the copies landed on the same layer as the original clip.

_That_ can work, and it can be done "relatively", but only as long as everything is meant to be staying on the same layer. Which, currently, it always is, because there's no way to trigger a paste with a layer number — the positioning doesn't exist, despite the `layer_id` argument to `Paste_Trigger`. That argument's effectively meaningless — which is a good thing, because if anyone tried to use it, even the old logic would break.

So the moment you move things off of their original layer, the information as to what layer they're _on_ is lost, and there's no way to recreate it. You can't store the layer number in relative terms because there's no such thing.

Reply:
Ok, so this optimization seems should go away, at least for the layers. But for position I think this is ok, right?